### PR TITLE
Created StudentHistory & Timeline component, added props to make it reusable

### PR DESCRIPTION
**Pull request by nondescryptid**
Screenshot 2020-10-17 at 4 03 56 PM

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -1,37 +1,13 @@
 <template>
```
Comment (lauratsui) [inline frontend/src/App.vue]: App.vue shouldn't be changed for now. Student profile component can eventually be rendered on the student profile page once that is merged.

File: frontend/src/main.js [hunk 2]
```diff
@@ -1,18 +1,11 @@
 import Vue from 'vue'
```
Comment (lauratsui) [inline frontend/src/main.js:2]: same comment as app.vue.

File: frontend/src/pages/StudentProfile.vue [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<template>
```
Comment (lauratsui) [inline frontend/src/pages/StudentProfile.vue]: Potentially conflicting with PR request to create separate student profile page #10

**Review by lauratsui — CHANGES_REQUESTED** [review]
Comments included in files. All the deleted files were accidentally deleted, suggest to reinstate.

File: frontend/src/pages/StudentProfile.vue (hunk 3 above)
Comment (lauratsui) [reply frontend/src/pages/StudentProfile.vue]: Also noticed the red border in the description class, could eventually be removed.

File: frontend/src/components/StudentHistory.vue [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+</script>
+
+<style>
+.card {
+    position:absolute;
+    width: 448px;
+    height: 600px;
+    /* From Figma. Commented for now, or else the card goes really far to the bottom-right.
+    left: 968px;
+    top: 276px;*/
```
Comment (thchia) [inline frontend/src/components/StudentHistory.vue]: I think all the styles for `.card` from here up are not really needed. The generated styles in Figma don't always make sense (that's why we still need devs 😈).

File: frontend/src/components/StudentHistory.vue [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    padding: 32px;
+}
+header {
+    padding-right: 350px;
```
Comment (thchia) [inline frontend/src/components/StudentHistory.vue]: I don't think this padding is needed. Text will be left-aligned by default for most of our browsers.

File: frontend/src/components/StudentHistory/Timeline.vue [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    height: 20px;
+    
+}
+#description {
+   margin-top: 8px;
+   margin-bottom: 8px;
+   margin-right: 24px;
+   margin-left: 9px;
+   width: 225px;
+   height: 40px;
```
Comment (thchia) [inline frontend/src/components/StudentHistory/Timeline.vue]: Do we need an explicit height here? The text might break into 2 lines (e.g. if a name is long).

**Review by thchia — CHANGES_REQUESTED** [review]
Overall, looks great! Not an easy piece to implement. Some small comments on the inclusion of Figma styles which are not always relevant/correct.

File: frontend/src/components/StudentHistory/Timeline.vue (hunk 6 above)
Comment (nondescryptid) [reply frontend/src/components/StudentHistory/Timeline.vue]: there's an explicit height as the vertical line running through the dots depends on the height of the elements being the same — as opposed to one big line running through all entries, each timeline item has a small line running through the dot, which joins together with other lines for other timeline items to form one long line. 
Guess we wouldn't need explicit height if the lines were dynamic, but I haven't figured out how.

File: frontend/src/pages/StudentProfile.vue (hunk 3 above)
Comment (nondescryptid) [reply frontend/src/pages/StudentProfile.vue]: I've deleted the StudentProfile.vue file!

File: frontend/src/App.vue (hunk 1 above)
Comment (nondescryptid) [reply frontend/src/App.vue]: Thanks Laura - It was originally like that for me to see it render, but i forgot to update it (git merge didn't work, but I've added all the files manually).

**Review by aelbore — APPROVED** [review]
looks good

**Review by thchia — APPROVED** [review]

**Comment by thchia** [comment]
Needs merge conflict resolution. Perhaps that might be something for the weekend, unless you are confident to try it yourself.

**Comment by nondescryptid** [comment]
I've resolved the conflict in package.json — I had added "run" to the serve script earlier while I was working on it  (was adding "run" by myself before Thomas' PR was approved and merged) although it was not needed ("run" was added to postInstall). 

Now that the conflict's resolved, can I merge the PR, or should I get a third party look it over and then merge it? thank you!! ☺️

**Comment by thchia** [comment]
@szktm in terms of resolving the issues with accidentally deleted/changed files, everything looks good. Haven't pulled and run it locally, but if you have addressed the previous comments by myself and Laura then I think we're good to go!

**Review by thchia — APPROVED** [review]

**Pull request merged**